Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -3502,22 +3518,65 @@ def qradar_search_retrieve_events_command(
-
+    calling_context = demisto.callingContext.get('context', {})
+    sm = get_schedule_metadata(context=calling_context)
+    end_date: datetime | None = dateparser.parse(sm.get('end_date'))
+    if not end_date or end_date.year == 1:
+        end_date = None
+    # determine if this is the last run of the polling command
+    is_last_run = (datetime.now() + timedelta(seconds=interval_in_secs)).timestamp() >= end_date.timestamp() \
+        if end_date else False
     events, status = poll_offense_events(client, search_id, should_get_events=True, offense_id=args.get('offense_id', ''))
+    if is_last_run and args.get('success') and not events:
+        # if last run, we want to get the events that were fetched in the previous calls
+        return CommandResults(
+            readable_output='Not all events were fetched. partial data is available.',
+        )
+
     if status == QueryStatus.ERROR.value:
         raise DemistoException('Polling for events failed')
     if status == QueryStatus.SUCCESS.value:
-        return CommandResults(outputs_prefix='QRadar.SearchEvents',
-                              outputs_key_field='ID',
-                              outputs={'Events': events, 'ID': search_id},
-                              readable_output=tableToMarkdown(f'Events returned from search_id {search_id}',
-                                                              events,
-                                                              )
-                              )
+        # return the result only if the all events were retrieved, unless for the last call for this function
+        offense_id = args.get('offense_id', '')
+        events_limit = int(args.get('events_limit', params.get('events_limit')))
+        fetch_mode: FetchMode = args.get('fetch_mode', params.get('fetch_mode'))
+        if argToBoolean(args.get('retry_if_not_all_fetched', False)) and  \
+            not is_last_run and not \
+                is_all_events_fetched(client, fetch_mode, offense_id, events_limit, events):
+            # return scheduled command result without search id to search again
+            polling_args = {
+                'interval_in_seconds': interval_in_secs,
+                'success': True,
+                **args
+            }
+
+            scheduled_command = ScheduledCommand(
+                command='qradar-search-retrieve-events',
+                next_run_in_seconds=interval_in_secs,
+                args=polling_args,
+            )
+            return CommandResults(scheduled_command=scheduled_command if is_polling else None,
+                                  readable_output='Not all events were fetched. Searching again.',
+                                  outputs_prefix='QRadar.SearchEvents',
+                                  outputs_key_field='ID',
+                                  outputs={'Events': events, 'ID': search_id, 'Status': QueryStatus.PARTIAL},
+                                  )
+
+        else:
```

Comment:
If you switch the if / else conditions you can entirely remove the `else`

Reply:
i can remove the else either way

Reply:
true :)